lib/krikri/map_crosswalk.rb

```diff
@@ -0,0 +1,234 @@
+      #   instead.
+      def build_aggregation
+        raise 'Tried to index a blank node!' if @parent.node?
+        hash.merge!({ :ingestType => 'item',
+                     :ingestionSequence => 1,
+                     :ingestDate => Date.today.as_json,
+                     :@context => "http://dp.la/api/items/context",
+                     :aggregatedCHO => '#sourceResource',
+                     :@id => make_id,
+                     :id => local_name
+                   })
+
+        set_value(hash, :dataProvider, @parent.dataProvider.map(&:label), true)
+
+        set_value(hash, :originalRecord, @parent.originalRecord, true) do |o|
+          o.rdf_subject.to_s
+        end
+
+        set_value(hash, :intermediateProvider,
+                  @parent.intermediateProvider,
+                  true) do |ip|
+          ip.label.any? ? ip.label : ip.providedLabel
+        end
+
+        hash[:sourceResource] = build_source_resource
+
+        set_value(hash, :provider, @parent.provider) do |provider|
+          build_provider(provider)
+        end
+
+        set_value(hash, :hasView, @parent.hasView, true) do |view|
+          view.rdf_subject.to_s
+        end
+
+        set_value(hash, :isShownAt, @parent.isShownAt, true) do |view|
+          view.rdf_subject.to_s
+        end
+
+        set_value(hash, :object, @parent.object, true) do |view|
+          view.rdf_subject.to_s
+        end
+
+        # This doesn't work, because we have no good way of implementing
+        #`#get_provider_id`.
+        #
+        # p = @parent.provider.first || DPLA::MAP::Agent.new
+        # p_id = p.node? ? p.rdf_subject.to_s.split('/').last : p.rdf_label
+        # hash[:_id] = "#{p_id}-#{get_provider_id}"
+        #
+        # instead... do:
+        hash[:_id] = hash[:id]
+
+        hash
+      end
+
+      def build_source_resource
+        return nil if @parent.sourceResource.empty?
+        parent_sr = @parent.sourceResource.first
+        sr = { :@id => "#{make_id}#sourceResource" }
+
+        set_value(sr, :contributor, parent_sr.contributor) do |contrib|
+          contrib.label.any? ? contrib.label : contrib.providedLabel
+        end
+
+        set_value(sr, :creator, parent_sr.creator) do |creator|
+          creator.label.any? ? creator.label : creator.providedLabel
+        end
+
+        set_value(sr, :description, parent_sr.description, true)
+        set_value(sr, :format, parent_sr.dcformat)
+        set_value(sr, :identifier, parent_sr.identifier)
+
+        set_value(sr, :language, parent_sr.language) do |lang|
+          lang.prefLabel.any? ? lang.prefLabel : lang.providedLabel
+        end
+
+        set_value(sr, :publisher, parent_sr.publisher) do |pub|
+          pub.label.any? ? pub.label : pub.providedLabel
+        end
+        set_value(sr, :relation, parent_sr.relation)
+
+        set_value(sr, :genre, parent_sr.genre) do |genre|
+          genre.prefLabel.any? ? genre.prefLabel : genre.providedLabel
```

Comment:
For `genre` and others, the expectation that it's a graph is making the mapping somewhat brittle, leading to errors like `Indexing error: undefined method `prefLabel' for "Photograph; b&w":String`. I'm guessing we should probably have some logic to handle strings, even though we know that the longer term goal is to have these properties behave like we expect.